2022/Q2/artifacts/pos-model/PoS-model-cubic.md

```diff
@@ -0,0 +1,493 @@
+## Assumptions/simplifications
+
+- There is a single token type
+
+## Data types
+
+```go
+type Addr
+type Key
+type Epoch uint
+type VotingPower uint
+
+type Validator struct {
+  consensus_key map<Epoch, Key>
+  state map<Epoch, {inactive, candidate, frozen, jailed}>
+  total_deltas map<Epoch, amount:int>
+  voting_power map<Epoch, VotingPower>
+  reward_address Addr
+}
+
+type Bond struct {
+  validator Addr //not used
+  source Addr //not used
+  deltas map<Epoch, int>
+  }
+
+type Unbond struct {
+  validator Addr //not used
+  source Addr //not used
+  deltas map<(start:Epoch, end:Epoch), int>
+}
+
+type Slash struct {
+  epoch Epoch
+  validator Addr
+  voting_power VotingPower //new in cubic slashing
+  slash_type {duplicate_vote, ligth_client_attack} //not used in cubic slashing
```

Comment:
I might have misunderstood, but I think we still have different types of misbehavior

Reply:
This is something I want to clarify. In the spec, the infraction type is never used explicitely.